File: database/src/verify-user.ts
import prisma from './database';

async function verifyUserData() {
  try {
    console.log('🔍 Verifying updated user data...\n');

    const user = await prisma.user.findFirst({
      where: { email: 'john.doe@example.com' },
      include: {
        creditCards: true,
        transactions: {
          take: 3,
          orderBy: { transactionDate: 'desc' }
        },
        alertRules: true
      }
    });

    if (!user) {
      console.log('❌ User not found');
      return;
    }

    console.log('👤 USER PROFILE:');
    console.log(`   Name: ${user.firstName} ${user.lastName}`);
    console.log(`   Email: ${user.email}`);
    console.log(`   Phone: ${user.phoneNumber}`);
    console.log(`   Active: ${user.isActive}\n`);

    console.log('📍 ADDRESS:');
    console.log(`   Street: ${user.addressStreet}`);
    console.log(`   City: ${user.addressCity}, ${user.addressState} ${user.addressZipCode}`);
    console.log(`   Country: ${user.addressCountry}\n`);

    console.log('💰 FINANCIAL INFO:');
    console.log(`   Credit Limit: $${user.creditLimit?.toString()}`);
    console.log(`   Current Balance: $${user.currentBalance?.toString()}\n`);

    console.log('📱 MOBILE APP LOCATION (Privacy Consented):');
    console.log(`   Consent Given: ${user.locationConsentGiven}`);
    if (user.locationConsentGiven) {
      console.log(`   Last Location: ${user.lastAppLocationLatitude}, ${user.lastAppLocationLongitude}`);
      console.log(`   Timestamp: ${user.lastAppLocationTimestamp?.toISOString()}`);
      console.log(`   Accuracy: ${user.lastAppLocationAccuracy}m\n`);
    }

    console.log('🛒 LAST TRANSACTION LOCATION:');
    console.log(`   Coordinates: ${user.lastTransactionLatitude}, ${user.lastTransactionLongitude}`);
    console.log(`   Location: ${user.lastTransactionCity}, ${user.lastTransactionState}, ${user.lastTransactionCountry}`);
    console.log(`   Timestamp: ${user.lastTransactionTimestamp?.toISOString()}\n`);

    console.log('💳 RELATED DATA:');
    console.log(`   Credit Cards: ${user.creditCards.length}`);
    console.log(`   Recent Transactions: ${user.transactions.length}`);
    console.log(`   Alert Rules: ${user.alertRules.length}`);

    console.log('\n✅ User data verification complete!');

  } catch (error) {
    console.error('❌ Error verifying user data:', error);
    throw error;
  } finally {
    await prisma.$disconnect();
  }
}

verifyUserData(); 

File: packages/db/src/db/scripts/verify_user.py
"""Verify user data by printing joined info"""

import asyncio

from sqlalchemy import select
from sqlalchemy.orm import selectinload

from db.database import SessionLocal
from db.models import User


async def verify() -> None:
    async with SessionLocal() as session:
        result = await session.execute(
            select(User)
            .options(
                selectinload(User.creditCards),
                selectinload(User.transactions),
                selectinload(User.alertRules),
            )
            .where(User.email == "john.doe@example.com")
        )
        user = result.scalar_one_or_none()
        if not user:
            print("User not found")
            return

        print("USER PROFILE:")
        print(f"   Name: {user.first_name} {user.last_name}")
        print(f"   Email: {user.email}")
        print(f"   Phone: {user.phone_number}")
        print(f"   Active: {user.is_active}\n")

        print("ADDRESS:")
        print(f"   Street: {user.address_street}")
        print(f"   City: {user.address_city}, {user.address_state} {user.address_zipcode}")
        print(f"   Country: {user.address_country}\n")

        print("FINANCIAL INFO:")
        print(f"   Credit Limit: {user.credit_limit}")
        print(f"   Current Balance: {user.credit_balance}\n")

        print("RELATED DATA:")
        print(f"   Credit Cards: {len(user.creditCards)}")
        print(f"   Transactions: {len(user.transactions)}")
        print(f"   Alert Rules: {len(user.alertRules)}")


if __name__ == "__main__":
    asyncio.run(verify())


